src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,475 @@
+     * Similar to the Mode settings in genIDauto() method. d:
+     *
+     * <p>
+     * DIGIT: {0123456789} Digit values only.</p>
+     * <p>
+     * LOWERCASE: {abc...xyz} Lowercase letters only. </p>
+     * <p>
+     * UPPERCASE: {ABC...XYZ} Uppercase letters only.</p>
+     * <p>
+     * MIXCASE: {abc...XYZ} Lowercase and Uppercase letters only. </p>
+     * <p>
+     * EXTENDED: {012...xyz} Digit values and Lowercase letters only. </p>
+     * <p>
+     * ALPHANUMERIC: {012...abc...XYZ} Digit values, Lowercase and Uppercase
+     * letters.</p>
+     *
+     * @param token: There are 2 different modes to choose from. RANDOM: Value
+     * by value, each value's order is randomized. SEQUENTIAL: Value by value,
+     * each value's order is sequenced.
+     * @return - The method checks to see if it is even possible to produce the
+     * requested amount of ids using the given parameters. If it can't it'll
+     * return an error message. Otherwise a reference to a JsonObject that
+     * contains Json list of ids is returned.
+     *
+     */
+    public String genIdAuto(String token) {
+
+        // checks to see if its possible to produce or add requested amount of
+        // ids to database
+        if (isPrefixAvailable(true, token)) {
+            // creates a list to hold redundant ids
+            Set<String> redundantIdList = new HashSet();
+
+            // a range of characters that can be used, retrieved by tokenMap
+            String characters = (String) this.tokenMaps.get(token);
+
+            // sets the number of ids to make
+            int numIdsToMake = this.AMOUNT;
+            int prefixSize = this.PREFIX.length();
+            int charactersLength = characters.length();
+
+            do {
+
+                // creates tentative ids that'll be checked for uniqueness
+                while (ID_LIST.size() < AMOUNT) {
+                    StringBuilder buffer = new StringBuilder();
+                    for (int i = 0; i < this.LENGTH - prefixSize; i++) {
+                        double index = Math.random() * charactersLength;
+                        buffer.append(characters.charAt((int) index));
+                    }
+                    String temp = buffer.toString();
+                    this.ID_LIST.add(this.PREFIX + temp);
+                }
+                redundantIdList.addAll(DATABASE_MANAGER.checkId(ID_LIST));
+
+                // removes any redundant ids
+                removeRedundantId(redundantIdList);
+
+                // if no redundant ids were found, add new ids to database
+                if (ID_LIST.size() == this.AMOUNT) {
+                    DATABASE_MANAGER.addId(ID_LIST);
+                } 
+                    // decreases the number of ids to make based on what was 
+                    // just added to ID_LIST
+                    numIdsToMake = this.AMOUNT - ID_LIST.size();
+                
+                System.out.println("numIdsToMake = " + numIdsToMake);

```

Comment:
Proper logging (using a Logback/SLF4J `Logger`) should be used here (and anywhere prints are used)